packages/angular-rx-store-example-counter/src/app/store/operators/scan-sum.ts

```diff
@@ -0,0 +1,13 @@
+import { Observable } from 'rxjs';
+import { scan, startWith } from 'rxjs/operators';
+
+/**
+ * An operator that sums up the values the source observables emits
+ * and emits the running total each time the source observable emits. Starts
+ * with emitting 0 when subscribed to.
+ */
+export const scanSum = () => (source$: Observable<any>): Observable<number> =>
+  source$.pipe(
+    scan((acc, val) => acc + val, 0),
+    startWith(0)
```

Comment:
is there a particular reason why we're using a `startWith(0)` here when it's being attached to a `BehaviorSubject` with a starting value of 0?

Reply:
`BehaviorSubject` with a starting value of 0 is used only for `count$`
and `localCount$` is obtained from `counterChange$.pipe(scanSum())`
that is why in `scanSum` was added operator `startWith(0)`